docs/getting-started.md

```diff
@@ -0,0 +1,449 @@
+ 
+# builds ostree + any additional commands
+COPY ostree-build.sh /
+
+# entry into the container will start at this directory
+WORKDIR /ostree
+
+# run the following as `/bin/sh -c` 
+# or enter the container to execute ./ostree-build.sh
+RUN ../ostree-build.sh
+
+```
+
+Save the following bash scripts in the same directory as the Dockerfile. Then change the mode bit of these files so that they are executable, by running `chmod +x ostree-git.sh ostree-build.sh`
+
+```bash
+#!/bin/bash
+
+# ostree-git.sh
+# Clone ostree and update master branch
+
+set -euo pipefail
+
+# Set $username to your GitHub username here.
+username=
+
+# clone your fork of the OSTree repo, this will be in the "/" directory
+git clone https://github.com/$username/ostree.git
+cd ostree
+
+# Add upstream as remote and update master branch
+git checkout master  
+git remote add upstream https://github.com/ostreedev/ostree.git
+git pull --rebase upstream master
+```
+
+```bash
+#!/bin/bash
+
+# ostree-build.sh
+# Build and test OSTree
+
+set -euo pipefail
+
+# $remote_branch is the name of the remote branch in your
+# repo that contains changes (e.g. my-patch).
+remote_branch=
+
+# fetch updates from origin
+# origin url should be your forked ostree repository
+git fetch origin
+
+# go to branch with changes
+# if this branch already exists then checkout that branch
+exit_code="$(git checkout --track origin/$remote_branch; echo $?)"
+if [[ "$exit_code" == 1 ]]
+then
+    echo "This branch:" $remote_branch "is not a remote branch"
+    exit
+fi
+
+# make sure branch with changes is up-to-date
+git pull origin $remote_branch
+
+# build OSTree commands for Fedora 28 and Fedora 28 Atomic Host
+./autogen.sh --prefix=/usr --libdir=/usr/lib64 --sysconfdir=/etc
+./configure --prefix=/usr
+make -j$((2 * $(nproc)))
+make install
+
+# any additional commands go here
+```
+
+**Build the container**
+
+Run `docker build` in the same directory of the `Dockerfile` like so:
+
+```bash
+$ docker build -t ostree-fedora-test .
+```
+
+When this build is done, the `-t` option tags the image as `ostree-fedora-test`.  
+
+**Note**: Do not forget the dot **.** at the end of the above command which specifies the location of the Dockerfile.
+
+You will see `ostree-fedora-test` listed when running `docker images`:
+
+```bash
+REPOSITORY                                 TAG                 IMAGE ID            CREATED             SIZE
+ostree-fedora-test                         latest              817c04cc3656        1 day ago          978MB
+```
+
+**Entering the Container**
+
+To **start** the `ostree-fedora-test` container, run:
+
+```bash
+$ docker run -it --rm --entrypoint /bin/sh --name ostree-testing ostree-fedora-test
+```
+
+**Note**:
+
+`--rm` option removes the name of the container i.e. `ostree-testing` whenever the container's shell prompt `#` exits. This is so that the container name `ostree-testing` can be re-used, otherwise remove it manually by running `docker rm <container name>`.
```

Comment:
To be a bit more precise, the `--rm` flag cleans up the container and container file system when the container exits.

https://docs.docker.com/engine/reference/run/#clean-up---rm